### PR TITLE
fix: add new types to udaf functions

### LINKED ISSUE #7787: EARLIEST_BY_OFFSET not supporting timestamp type
**Describe the bug**
Version: 0.17
EARLIEST_BY_OFFSET(<some_timestamp_column>) throws an error.

**To Reproduce**
Create any query with aggregation that tries to aggregate timestamp fields.

**Expected behavior**
no error

**Actual behaviour**
ERROR: Function 'EARLIEST_BY_OFFSET' does not accept parameters (TIMESTAMP).

### PR DESCRIPTION
**Pull request by jzaralim**
### Description 
Adds support for timestamp, time, date and bytes to collect_list, collect_set, latest_by_offset and earliest_by_offset.

Fixes #7787

### Testing done 
unit/qtt

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-functional-tests/src/test/resources/query-validation-tests/collect-list.json [hunk 1]
```diff
@@ -108,6 +108,76 @@
...
+      ],
+      "outputs": [
+        {"topic": "S2", "key": 0, "value": {"COLLECTED":[10]}},
+        {"topic": "S2", "key": 0, "value": {"COLLECTED":[10,20]}},
+        {"topic": "S2", "key": 0, "value": {"COLLECTED":[10,20,30]}}
+      ]
+    },
+    {
+      "name": "collect_list bytes map",
+      "format": ["JSON"],
```
Comment (vcrfxia) [inline ksqldb-functional-tests/src/test/resources/query-validation-tests/collect-list.json:167]: Out of curiosity, does the PROTOBUF format not support the BYTES type?

**Review by vcrfxia — APPROVED** [review]
Thanks @jzaralim ! LGTM with a couple minor notes inline.

File: ksqldb-engine/src/main/java/io/confluent/ksql/function/udaf/offset/LatestByOffset.java [hunk 2]
```diff
@@ -188,6 +196,114 @@ private LatestByOffset() {
...
+    return latestTimestamp(true);
+  }
+
+  @UdafFactory(description = "return the latest value of a timestamp column",
+      aggregateSchema = "STRUCT<SEQ BIGINT, VAL TIMESTAMP>")
+  public static Udaf<Timestamp, Struct, Timestamp> latestTimestamp(final boolean ignoreNulls) {
+    return latest(STRUCT_TIMESTAMP, ignoreNulls, getComparator(ignoreNulls));
+  }
+
+  @UdafFactory(description = "return the latest N value of a timestamp column",
```
[the remark this answers is not in the document]
Comment (jzaralim) [reply ksqldb-engine/src/main/java/io/confluent/ksql/function/udaf/offset/LatestByOffset.java]: Done

File: ksqldb-functional-tests/src/test/resources/query-validation-tests/collect-list.json (hunk 1 above)
Comment (jzaralim) [reply ksqldb-functional-tests/src/test/resources/query-validation-tests/collect-list.json:167]: It does, it just requires specifying the schema to the test, which I was too lazy to do.

**Pull request merged**